connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -237,7 +246,15 @@ object KustoWriter {
         val blobPath = blobUri + sas
         val blobSourceInfo = new BlobSourceInfo(blobPath, size)
 
-        partitionsResults.add(PartitionResult(ingestClient.ingestFromBlob(blobSourceInfo, props), partitionId))
+        val res: IngestionResult = try {
+          KDSU.logInfo(myName, s"Queued blob for ingestion in partition ${TaskContext.getPartitionId} for requestId: '$requestId}")
+          ingestClient.ingestFromBlob(blobSourceInfo, props)
+        } catch {
```

Comment:
Just thinking aloud:
This approach seems best if we will only ever want 1 retry and only if it's a `IngestionServiceException`. But I'm thinking we may want to add more in the future, and I can imagine wanting to retry more than once if for certain types of transient errors (like throttling). OTOH, we don't want to overengineer for a case that isn't relevant.

Reply:
IngestionServiceException are all the transient errors from the service - we can rely on this conception it is - we might want to add this to documentation and code description but this is true